File: C/Ledger/ledger.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define MAX_ACCOUNTS 100
#define MAX_TRANSACTIONS 100

typedef struct {
    int account_number;
    char name[100];
    float balance;
    float transactions[MAX_TRANSACTIONS];
    int transaction_count;
} Account;

Account accounts[MAX_ACCOUNTS];
int num_accounts = 0;

void createAccount(int account_number, char *name) {
    if (num_accounts >= MAX_ACCOUNTS) {
        printf("Maximum accounts reached!\n");
        return;
    }

    accounts[num_accounts].account_number = account_number;
    strcpy(accounts[num_accounts].name, name);
    accounts[num_accounts].balance = 0;
    accounts[num_accounts].transaction_count = 0;
    num_accounts++;
}

void deposit(int account_number, float amount) {
    for (int i = 0; i < num_accounts; i++) {
        if (accounts[i].account_number == account_number) {
            accounts[i].balance += amount;
            accounts[i].transactions[accounts[i].transaction_count++] = amount;
            printf("Deposit successful!\n");
            return;
        }
    }
    printf("Account not found!\n");
}

void withdraw(int account_number, float amount) {
    for (int i = 0; i < num_accounts; i++) {
        if (accounts[i].account_number == account_number) {
            if (amount > accounts[i].balance) {
                printf("Insufficient funds!\n");
                return;
            }
            accounts[i].balance -= amount;
            accounts[i].transactions[accounts[i].transaction_count++] = -amount;
            printf("Withdrawal successful!\n");
            return;
        }
    }
    printf("Account not found!\n");
}

void checkBalance(int account_number) {
    for (int i = 0; i < num_accounts; i++) {
        if (accounts[i].account_number == account_number) {
            printf("Balance for account %d: %.2f\n", account_number, accounts[i].balance);
            return;
        }
    }
    printf("Account not found!\n");
}

void viewTransactions(int account_number) {
    for (int i = 0; i < num_accounts; i++) {
        if (accounts[i].account_number == account_number) {
            printf("Transaction history for account %d:\n", account_number);
            for (int j = 0; j < accounts[i].transaction_count; j++) {
                printf("%s %.2f\n", (accounts[i].transactions[j] > 0) ? "Deposit:" : "Withdrawal:", accounts[i].transactions[j]);
            }
            return;
        }
    }
    printf("Account not found!\n");
}

int main() {
    int choice;
    int account_number;
    float amount;
    char name[100];

    do {
        printf("\nBank Account Management System\n");
        printf("1. Create Account\n");
        printf("2. Deposit\n");
        printf("3. Withdraw\n");
        printf("4. Check Balance\n");
        printf("5. View Transactions\n");
        printf("6. Exit\n");
        printf("Enter your choice: ");
        scanf("%d", &choice);

        switch (choice) {
            case 1:
                printf("Enter account number: ");
                scanf("%d", &account_number);
                printf("Enter name: ");
                scanf("%s", name);
                createAccount(account_number, name);
                break;
            case 2:
                printf("Enter account number: ");
                scanf("%d", &account_number);
                printf("Enter amount to deposit: ");
                scanf("%f", &amount);
                deposit(account_number, amount);
                break;
            case 3:
                printf("Enter account number: ");
                scanf("%d", &account_number);
                printf("Enter amount to withdraw: ");
                scanf("%f", &amount);
                withdraw(account_number, amount);
                break;
            case 4:
                printf("Enter account number: ");
                scanf("%d", &account_number);
                checkBalance(account_number);
                break;
            case 5:
                printf("Enter account number: ");
                scanf("%d", &account_number);
                viewTransactions(account_number);
                break;
            case 6:
                printf("Exiting...\n");
                break;
            default:
                printf("Invalid choice!\n");
        }
    } while (choice != 6);

    return 0;
}

File: C#/Ledger/ledger.cs
using System;
using System.Collections.Generic;

class Transaction {
    public float Amount { get; }
    public string Type { get; }

    public Transaction(float amount, string type) {
        Amount = amount;
        Type = type;
    }
}

class Account {
    public int AccountNumber { get; }
    public string Name { get; }
    public float Balance { get; private set; }
    private List<Transaction> transactions;

    public Account(int accountNumber, string name) {
        AccountNumber = accountNumber;
        Name = name;
        Balance = 0;
        transactions = new List<Transaction>();
    }

    public void Deposit(float amount) {
        Balance += amount;
        transactions.Add(new Transaction(amount, "Deposit"));
        Console.WriteLine("Deposit successful!");
    }

    public void Withdraw(float amount) {
        if (amount > Balance) {
            Console.WriteLine("Insufficient funds!");
            return;
        }
        Balance -= amount;
        transactions.Add(new Transaction(amount, "Withdrawal"));
        Console.WriteLine("Withdrawal successful!");
    }

    public void CheckBalance() {
        Console.WriteLine($"Balance for account {AccountNumber}: {Balance}");
    }

    public void ViewTransactions() {
        Console.WriteLine($"Transaction history for account {AccountNumber}:");
        foreach (Transaction transaction in transactions) {
            Console.WriteLine($"{transaction.Type}: {transaction.Amount}");
        }
    }
}

class Program {
    static void Main(string[] args) {
        int choice;
        int accountNumber;
        float amount;
        string name;

        List<Account> accounts = new List<Account>();

        do {
            Console.WriteLine("\nBank Account Management System");
            Console.WriteLine("1. Create Account");
            Console.WriteLine("2. Deposit");
            Console.WriteLine("3. Withdraw");
            Console.WriteLine("4. Check Balance");
            Console.WriteLine("5. View Transactions");
            Console.WriteLine("6. Exit");
            Console.Write("Enter your choice: ");
            choice = int.Parse(Console.ReadLine());

            switch (choice) {
                case 1:
                    Console.Write("Enter account number: ");
                    accountNumber = int.Parse(Console.ReadLine());
                    Console.Write("Enter name: ");
                    name = Console.ReadLine();
                    accounts.Add(new Account(accountNumber, name));
                    break;
                case 2:
                    Console.Write("Enter account number: ");
                    accountNumber = int.Parse(Console.ReadLine());
                    Console.Write("Enter amount to deposit: ");
                    amount = float.Parse(Console.ReadLine());
                    foreach (Account account in accounts) {
                        if (account.AccountNumber == accountNumber) {
                            account.Deposit(amount);
                            break;
                        }
                    }
                    break;
                case 3:
                    Console.Write("Enter account number: ");
                    accountNumber = int.Parse(Console.ReadLine());
                    Console.Write("Enter amount to withdraw: ");
                    amount = float.Parse(Console.ReadLine());
                    foreach (Account account in accounts) {
                        if (account.AccountNumber == accountNumber) {
                            account.Withdraw(amount);
                            break;
                        }
                    }
                    break;
                case 4:
                    Console.Write("Enter account number: ");
                    accountNumber = int.Parse(Console.ReadLine());
                    foreach (Account account in accounts) {
                        if (account.AccountNumber == accountNumber) {
                            account.CheckBalance();
                            break;
                        }
                    }
                    break;
                case 5:
                    Console.Write("Enter account number: ");
                    accountNumber = int.Parse(Console.ReadLine());
                    foreach (Account account in accounts) {
                        if (account.AccountNumber == accountNumber) {
                            account.ViewTransactions();
                            break;
                        }
                    }
                    break;
                case 6:
                    Console.WriteLine("Exiting...");
                    break;
                default:
                    Console.WriteLine("Invalid choice!");
                    break;
            }
        } while (choice != 6);
    }
}
